INSTALL.md

```diff
@@ -1734,6 +1735,37 @@ to install additional support software to obtain a random seed and reseed
 the CSPRNG manually.  Please check out the manual pages for `RAND_add()`,
 `RAND_bytes()`, `RAND_egd()`, and the FAQ for more information.
 
+Notes on assembler modules compilation
+--------------------------------------
+
+Compilation of some code paths in assembler modules might depend on whether
+current assembler version supports certain ISA extension or not. Base minimum of
+AES-NI, PCLMULQDQ, SSSE3 and SHA extension code paths are always assembled.
+Apart from that, minimum assembler version requirements are summarized in below
+table:
+
+   Extension   | GNU as | nasm   | llvm
+   ------------+--------+--------+--------
+   AVX         | 2.19   | 2.09   | 3.0
+   AVX2        | 2.22   | 2.10   | 3.1
+   ADCX/ADOX   | 2.23   | 2.10   | 3.3
+   AVX512      | 2.25   | 2.11.8 | see llvm notes below
+   AVX512IFMA  | 2.26   | 2.11.8 | see llvm notes below
+   VAES        | 2.30   | 2.13.3 |
+
+Even though AVX512 support was implemented in llvm 3.6, compilation of
+assembly modules apparently requires explicit -march flag. But then
+compiler generates processor-specific code, which in turn contradicts
+the mere idea of run-time switch execution facilitated by the variable
+in question. Till the limitation is lifted, it's possible to work around
+the problem by making build procedure use following script:
+
+   #!/bin/sh
+   exec clang -no-integrated-as "$@"
```

Comment:
You'll probably need 4 spaces before as the formatting is not recognized https://github.com/openssl/openssl/blob/0b3ba222fd9b966d27236420080fcef11fe6dab7/INSTALL.md